bundles/org.insilico.sbmlsheets/src/org/insilico/sbmlsheets/core/CompartmentInstance.java

```diff
@@ -0,0 +1,417 @@
+    return sbmlCompartmentId;
+  }
+
+
+  public StringProperty outerCompartmentIdProperty() {
+    return outerCompartmentId;
+  }
+
+
+  public StringProperty unitProperty() {
+    return unit;
+  }
+
+
+  public StringProperty sizeProperty() {
+    return size;
+  }
+
+ 
+  
+  public StringProperty commentProperty() {
+    return comment;
+  }
+
+  public StringProperty referenceNameProperty() {
+    return referenceName;
+  }
+
+  public StringProperty referencePubMedProperty() {
+    return referencePubMed;
+  }
+
+  public StringProperty referenceDOIProperty() {
+    return referenceDOI;
+  }
+
+  public StringProperty descriptionProperty() {
+    return description;
+  }
+
+  public StringProperty miriamAnnotationsProperty() {
+    return miriamAnnotations;
+  }
+  public StringProperty typeProperty() {
+    return type;
+  }
+  public StringProperty symbolProperty() {
+    return symbol;
+  }
+  public StringProperty positionXProperty() {
+    return positionX;
+  }public StringProperty positionYProperty() {
+    return positionY;
+  }
+  
+  public StringProperty sboTermProperty() {
+    return sboTerm;
+  }public StringProperty identifiersProperty() {
+    return identifiers;
+  }
+  
+  public ObservableValue getProperty(String attribute) {
+    for (int j=0; j < attributeTypes.size(); j++) {
+      if (attributeTypes.get(j).split("_")[0].equals(attribute)) {
+        return attributes.get(j);
+      }
+    }
+    return id;
+  }
+
+  private void setProperDouble(StringProperty attribute, String newValue, String oldValue) {
+    if (newValue.matches("^[0-9]+\\.[0-9]+$")) {
+      attribute.set(newValue);
+    }else if (newValue.matches("^[0-9]+$")) {
+      attribute.set(newValue+".0");
+    }else {
+      if (oldValue.matches("^[0-9]+\\.[0-9]+$")) {
+        attribute.set(oldValue);
+      }else if (oldValue.matches("^[0-9]+$")) {
+        attribute.set(oldValue+".0");
+      }else {
+        attribute.set(defaultDouble);
+      }
+    }
+  }
+  
+  private void setProperInteger(StringProperty attribute, String newValue, String oldValue) {
+    if (newValue.matches("^[0-9]+\\.[0-9]+$")) {
+      attribute.setValue(newValue.split("\\.")[0]);
+    }else if (newValue.matches("^[0-9]+")) {
+      attribute.setValue(newValue);
+    }else {
+      if (oldValue.matches("^[0-9]+\\.[0-9]+$")) {
+        attribute.setValue(oldValue.split("\\.")[0]);
+      }else if (oldValue.matches("^[0-9]+$")) {
+        attribute.setValue(oldValue);
+      }else {
+        attribute.setValue(defaultInt);
+      }
+    }
+  }
+
+  private String getCorrectValue(String newValue, String type) {
+    String value = newValue.toString();
+    switch (type) {
+      case "Int":
+        System.out.println("changing to proper int");
+        if (value.matches("^[0-9]+\\.[0-9]+$")) {
+          value = value.split("\\.")[0];
+        }else if (value.matches("^[0-9]+$")) {
+          System.out.println("int stays int");
```

Comment:
I am not sure if it is necessary to provide this comment. What does "int stays int" says to the user during the normal usage of the application? Probably, logging this information could make sense (in proper way)?